pennylane/circuit_graph.py

```diff
@@ -12,305 +12,283 @@
         """
-
-        for idx, op in enumerate(queue + observables):
-            cmd = Command(
-                name=op.name, op=op, return_type=getattr(op, "return_type", None), idx=idx
-            )
-
+        for k, op in enumerate(ops):
+            op.queue_idx = k  # store the queue index in the Operation
             for w in set(op.wires):
-                if w not in self._grid:
-                    # wire is not already in the grid;
-                    # add the corresponding wire to the grid
-                    self._grid[w] = []
-
-                # Add the operation to the grid, to the end of the specified wire
-                self._grid[w].append(cmd)
-
-        self._graph = nx.DiGraph()
+                # Add op to the grid, to the end of wire w
+                self._grid.setdefault(w, []).append(op)
 
+        self._graph = nx.DiGraph()  #: nx.DiGraph: DAG representation of the quantum circuit
         # Iterate over each wire in the grid
-        for cmds in self._grid.values():
-            if cmds:
+        for wire in self._grid.values():
+            if wire:
                 # Add the first operation on the wire to the graph
                 # This operation does not depend on any others
-                self._graph.add_node(cmds[0].idx, **cmds[0]._asdict())
+                self._graph.add_node(wire[0])
 
-            for i in range(1, len(cmds)):
+            for i in range(1, len(wire)):
                 # For subsequent operations on the wire:
-                if cmds[i].idx not in self._graph:
+                if wire[i] not in self._graph:
                     # Add them to the graph if they are not already
                     # in the graph (multi-qubit operations might already have been placed)
-                    self._graph.add_node(cmds[i].idx, **cmds[i]._asdict())
+                    self._graph.add_node(wire[i])
 
-                # Create an edge between this operation and the
-                # previous operation
-                self._graph.add_edge(cmds[i - 1].idx, cmds[i].idx)
+                # Create an edge between this operation and the previous operation
+                self._graph.add_edge(wire[i - 1], wire[i])
 
     @property
     def observables(self):
-        """
-        Return a list of operations that have a return type.
-        """
-        return [node["op"] for node in self.observable_nodes]
+        """Observables in the circuit, sorted by queue index.
 
-    @property
-    def observable_nodes(self):
-        """
-        Return a list of nodes of operations that have a return type, sorted by "idx".
+        TODO or in any topological order?
+
+        Returns:
+            list[Observable]: observables
         """
-        nodes = [node for node in self.graph.nodes.values() if node["return_type"]]
-        return sorted(nodes, key=lambda node: node["idx"])
+        #nodes = [node for node in self._graph.nodes if isinstance(node, qml.operation.Observable)]
```

Comment:
Is this line intended to be removed?

Reply:
Yeah, maybe. I'm not quite sure what's the current distinction between Observables and non-Observables. Some Observables such as `PauliX` can represent both a gate and a measurement I think.

What this function should return is Observables with `return_type` set to anything else than None, right?